### PR TITLE
Standardize timeouts and remove sync over async in tests

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes flaky tests - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=977144&view=ms.vss-test-web.build-test-results-tab&runId=21008528&resultId=100302&paneView=debug

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
I was going to mention that it may be helpful to have a test, but I can't really figure out how we'd be able to reliably do so....

**Comment by JoshLove-msft** [comment]
> I was going to mention that it may be helpful to have a test, but I can't really figure out how we'd be able to reliably do so....

Heh yeah, we sort of do have tests for this which is how I noticed it.

**Comment by pakrym** [comment]
Is it possible that we will cancel the renewal too early with the processor still running?

**Comment by JoshLove-msft** [comment]
> Is it possible that we will cancel the renewal too early with the processor still running?

We close the receiver on the next line. At this point, all message handlers have finished processing.

**Review by pakrym — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs [hunk 1]
```diff
@@ -67,6 +69,7 @@ internal class ReceiverManager
             {
                 try
                 {
+                    _closingTokenSource.Cancel();
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs:72]: On further inspection, I don't think is needed. We already cancel the renew lock token in the finally clause of ProcessOneMessage.

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs]: I suspect the underlying issue is due to the very short lock duration these tests were using.

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs [hunk 2]
```diff
@@ -240,7 +238,7 @@ protected async Task ProcessOneMessageWithinScopeAsync(ServiceBusReceivedMessage
             }
             finally
             {
-                renewLockCancellationTokenSource?.Cancel();
+                await CancelTask(renewLockCancellationTokenSource, renewLock).ConfigureAwait(false);
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs:241]: Minor reshuffle

**Pull request merged**